specs-go/config.go

```diff
@@ -469,3 +479,77 @@ type Syscall struct {
 }
+
+// Windows contains platform specific configuration for Windows based containers.
+type Windows struct {
+	// Resources contains constraints for containers.
+	Resources *WindowsResources `json:"resources,omitempty"`
+	// Networking contains the network connection settings for containers.
+	Networking *Networking `json:"networking,omitempty"`
+	// FirstStart is an optimization on first boot of a container.
+	FirstStart bool `json:"first_start,omitempty"`
+	// LayerFolder is the path to the current layer folder.
+	LayerFolder string `json:"layer_folder,omitempty"`
+	// Layer contains the filepaths of the parent layers.
+	LayerPaths []string `json:"layer_paths,omitempty"`
+	// Settings relating to Hyper-V containers, omitted otherwise.
+	HvRuntime *HvRuntime `json:"hv_runtime,omitempty"`
+}
+
+// WindowsResources has container runtime resource constraints for Windows based containers.
+type WindowsResources struct {
+	// Network resource configuration.
+	Network *WindowsNetwork `json:"network,omitempty"`
+	// Memory resource configuration.
+	Memory *WindowsMemory `json:"memory,omitempty"`
+	// CPU resource configuration.
+	CPU *WindowsCPU `json:"cpu,omitempty"`
+	// Storage resource configuration.
+	Storage *WindowsStorage `json:"storage,omitempty"`
+}
+
+// WindowsNetwork has network runtime resource constraings for Windows based containers.
+type WindowsNetwork struct {
+	// Bandwidth is the maximum egress bandwidth in bytes per second.
+	Bandwidth *uint64 `json:"bandwidth,omitempty"`
+}
+
+// WindowsMemory has memory runtime resource constraints for Windows based containers.
+type WindowsMemory struct {
+	// Memory limit (in bytes).
+	Limit *int64 `json:"limit,omitempty"`
+	// Memory reservation (in bytes).
+	Reservation *uint64 `json:"reservation,omitempty"`
+}
+
+// WindowsCPU has CPU runtime resource constraints for Windows based containers.
+type WindowsCPU struct {
+	// Number of CPUs available to the container. This is an approximation for Windows Server Containers.
+	Count *uint64 `json:"count,omitempty"`
+	// CPU shares (relative weight (ratio) vs. other containers with cpu shares). Range is from 1 to 10000.
+	Shares *uint64 `json:"shares,omitempty"`
+	// Percent of available CPUs usable by the container.
+	Percent *int64 `json:"percent,omitempty"`
+}
+
+// WindowsStorage has storage resource constraints for Windows based containers.
+type WindowsStorage struct {
+	// Specifies maximum Iops for the system drive.
+	Iops *uint64 `json:"iops,omitempty"`
+	// Specifies maximum bytes per second for the system drive.
+	Bps *uint64 `json:"bps,omitempty"`
+	// Sandbox size indicates the size to expand the system drive to if it is currently smaller.
+	SandboxSize *uint64 `json:"sandbox_size,omitempty"`
+}
+
+// Networking contains the network configuration for Windows based containers.
+type Networking struct {
+	// List of endpoints to be attached to the container.
+	EndpointList []string `json:"endpoints,omitempty"`
```

Comment:
I'd expect `Endpoints` instead of `EndpointList` for the Go parameter behind and `endpoints` JSON parameter.